WebDriverAgentLib/Categories/XCUIElement+FBClassChain.m

```diff
@@ -0,0 +1,104 @@
+/**
+ * Copyright (c) 2015-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "XCUIElement+FBClassChain.h"
+#import "FBClassChainQueryParser.h"
+#import "FBElementUtils.h"
+#import "FBMacros.h"
+#import "XCElementSnapshot.h"
+#import "XCUIElement+FBUtilities.h"
+
+NSString *const FBClassChainQueryParseException = @"FBClassChainQueryParseException";
+
+@implementation XCUIElement (FBClassChain)
+
+- (NSArray<XCUIElement *> *)fb_descendantsMatchingClassChain:(NSString *)classChainQuery shouldReturnAfterFirstMatch:(BOOL)shouldReturnAfterFirstMatch
+{
+  NSError *error;
+  FBClassChain parsedChain = [FBClassChainQueryParser parseQuery:classChainQuery error:&error];
```

Comment:
This `NSArray` not `FBClassChain`. `FBClassChain` is undefined, I can't find any definition.